src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs

```diff
@@ -36,18 +37,24 @@ public TxReceipt Decode(RlpStream rlpStream, RlpBehaviors rlpBehaviors = RlpBeha
             }
             
             TxReceipt txReceipt = new TxReceipt();
-            rlpStream.ReadSequenceLength();
+            var lenght = rlpStream.ReadSequenceLength();
             byte[] firstItem = rlpStream.DecodeByteArray();
-            if (firstItem.Length == 1)
+            if (firstItem.Length == 1 && (firstItem[0] == 0 || firstItem[0] == 1))
```

Comment:
what is this? why 0 or 1?

Reply:
In order to support situation when we don't get StatusCode nor StateRoot